tests/foreman/destructive/test_rename.py

```diff
@@ -63,19 +63,22 @@ def test_positive_rename_satellite(module_org, module_product, module_target_sat
     username = settings.server.admin_username
     password = settings.server.admin_password
     old_hostname = module_target_sat.execute('hostname').stdout.strip()
-    new_hostname = f'new-{old_hostname}'
+    old_shortname, old_domain = old_hostname.split(".", 1)
+    new_hostname = f'{old_shortname}-changed.{old_domain}'
     # create installation medium with hostname in path
     medium_path = f'http://{old_hostname}/testpath-{gen_string("alpha")}/os/'
     medium = module_target_sat.api.Media(organization=[module_org], path_=medium_path).create()
     repo = module_target_sat.api.Repository(product=module_product, name='testrepo').create()
     # create /etc/hosts entry to pass s-c-h validation
-    sat_ip = module_target_sat.execute(
-        "ip addr show eth0 | grep 'inet ' | awk '{print $2}' | cut -d/ -f1"
-    ).stdout.strip()
+    sat_ip, sat_prefix = (
+        module_target_sat.execute("ip --oneline addr show scope global | awk '{print$4}'")
+        .stdout.strip()
+        .split('/')
+    )
     module_target_sat.execute(f'echo "{sat_ip} {old_hostname} {new_hostname}" >> /etc/hosts')
```

Comment:
This really isn't correct IMHO because it can return multiple IPs. I think the following implementation should be safe and reliable
```suggestion
    for line in module_target_sat.execute("ip --oneline addr show scope global"):
        if line:
            sat_ip, _prefix = line[3].split('/')
            module_target_sat.execute(f'echo "{sat_ip} {old_hostname} {new_hostname}" >> /etc/hosts')
```

You can also consider a line to delete the IP line: `sed -i '/{sat_ip}/d /etc/hosts`

Reply:
Cool cool but lets not handle dualstack in this PR. The dualstack jobs are not yet even triggerred.

Reply:
Why shouldn't we fix this? You're rewriting it and why would you choose to implement it partially? Note that single stack with multiple NICs you can also get multiple IPs back.

Reply:
Dualstack support added